cmd/admin-heal-ops.go

```diff
@@ -531,6 +535,29 @@ func (h *healSequence) traverseAndHeal() {
 	close(h.traverseAndHealDoneCh)
 }
 
+// healConfig - heals config.json, retrun value indicates if a failure occurred.
+func (h *healSequence) healConfig() error {
+	// Get current object layer instance.
+	objectAPI := newObjectLayerFn()
+	if objectAPI == nil {
+		return errServerNotInitialized
+	}
+
+	configFile := path.Join(minioConfigPrefix, minioConfigFile)
+	configBackupFile := path.Join(minioConfigPrefix, minioConfigBackupFile)
+	for _, cfg := range []string{configFile, configBackupFile} {
+		res, err := objectAPI.HealObject(h.ctx, minioMetaBucket, cfg, h.settings.DryRun)
+		if err != nil {
+			return err
+		}
+		res.Type = madmin.HealItemBucketMetadata
```

Comment:
It seems config is not exactly a bucket metadata, let's add deployment metadata if possible.

Reply:
Will add it later, for now it is repurposed in this manner. It requires a release of mc etc to get this all working which I didn't see as a good idea.

Reply:
> Will add it later, for now it is repurposed in this manner. It requires a release of mc etc to get this all working which I didn't see as a good idea.

okay, one last thing.. it seems that config backup is not there, and I guess we should not return an error if it is not found

Reply:
Does it return an error ?

Reply:
> Does it return an error ?

Yes, this is what mc shows currently
```
Heal had an error -  Object not found: .minio.sys#config/config.json.backup
```

Reply:
Looks like that should be ignored by HealObject anycase @vadmeste